### ISSUE TITLE
Find a maximum matching of a non-bipartite graph

### ISSUE THREAD

**Issue by lichengzhang1** [igraph/igraph#2403]
A maximum matching (also known as maximum-cardinality) matching is a matching that contains the largest possible number of edges. Now, _igraph/c_ only has `igraph_maximum_bipartite_matching`, not for non-bipartite graphs.

**References**

1. https://en.wikipedia.org/wiki/Blossom_algorithm

**Comment by ntamas** [comment]
I've been wanting to add this for a long time now so I totally support the idea.

**Comment by szhorvat** [comment]
IGraph/M has this borrowed from [LEMON](https://lemon.cs.elte.hu/trac/lemon), it'd be great to have a pure-igraph implementation.

**Comment by szhorvat** [comment]
You can work on it, but note that our capacity to help is very limited at the moment. There is a large backlog of other PRs to deal with first.

Here's a quickstart guide: https://github.com/igraph/igraph/wiki/Quickstart-for-new-contributors There is more useful info in the wiki: https://github.com/igraph/igraph/wiki

You should target the PR on the `develop` branch if you start right now (because `develop` will be merged into `master` quite soon).

Before starting any coding, it would be best to give a short summary of the algorithm described in the paper, with references to the relevant parts of the paper. I also strongly suggest reviewing the literature, especially recent papers, to see if this is the best algorithm to implement. A quick search seems to indicate that the algorithm you reference is not simple to implement. Perhaps there are newer and simpler algorithms that are equally fast?

**Comment by EdgarAlenPoe** [comment]
I've decided the algorithm I considered before was too difficult, and will be going with the edmonds blossom algorithm.
I've closed my previous pull request and will open a new one soon.
